routing/bicycle_directions.cpp

```diff
@@ -170,7 +185,13 @@ void BicycleDirectionsEngine::Generate(IRoadGraph const & graph, vector<Junction
   Route::TTimes turnAnnotationTimes;
   Route::TStreets streetNames;
 
-  MakeTurnAnnotation(resultGraph, delegate, routeGeometry, turns, turnAnnotationTimes, streetNames);
+  MakeTurnAnnotation(resultGraph, delegate, routeGeometry, turns, turnAnnotationTimes,
+                     streetNames, routeSegs);
+  // @TODO(bykoianko) The invariant below it's an issue but now it's so and it should be checked.
+  // The problem is every edge is added as a pair of point to route geometry.
+  // So all the points except for beginning and ending are duplicated. It should
+  // be fixed in the future.
+  ASSERT_EQUAL(routeGeometry.size(), 2 * routeSegs.size(), ());
```

Comment:
It depends on what do you mean by saying an edge.

At this moment the route for all routing except for osrm looks like:
```
 0----1    4----5
      2----3    6---7
```
This route composes of 4 connected routing::Edge(s). That means that point 1 and 2, 3 and 4 or 5 and 6 have the same coords. So I put 4 items in routeSegs. More then that the route comes from the router as 5 point route. But then in Bicycle(Pedestrian)DirectionsEngine::Generate() the path converted to 8-point route with 3 zero length segments.

I guess the route in this example should contain only 5 points and it's an issue that after some manipulations it turns to 8 point route. I consider this turning as an issue.

I try to git rid of this effect in the PR but face some further issues. I guess it should be refactored when we try to get rid of ReconstractRoute/ReconstructPath/RedressRoute and so on. This problems are close to each other.